src/sdi-notify.c

```diff
@@ -294,8 +298,8 @@ static void show_pending_update_notification(SdiNotify *self,
   if (allow_to_ignore) {
     g_autoptr(GVariantBuilder) builder =
         g_variant_builder_new(G_VARIANT_TYPE("as"));
-    for (; snaps != NULL; snaps = snaps->next) {
-      SnapdSnap *snap = (SnapdSnap *)snaps->data;
+    for (int i = 0; i != g_list_model_get_n_items(snaps); i++) {
```

Comment:
nit: Conventionally this would be `<` instead of `!=`

Reply:
Yes, you are right. Fixed.